src/libraries/System.Threading.ThreadPool/tests/ThreadPoolTests.cs

```diff
@@ -1020,6 +1021,65 @@ public static void CooperativeBlockingWithProcessingThreadsAndGoalThreadsAndAddW
             }).Dispose();
         }
 
+        [ConditionalFact(nameof(IsThreadingAndRemoteExecutorSupported))]
+        public void FileStreamFlushAsyncThreadPoolDeadlockTest()
+        {
+            // This test was occasionally causing the deadlock described in https://github.com/dotnet/runtime/pull/68171. Run it
+            // in a remote process to test it with a dedicated thread pool.
+            RemoteExecutor.Invoke(async () =>
+            {
+                const int OneKibibyte = 1 << 10;
+                const int FourKibibytes = OneKibibyte << 2;
+                const int FileSize = 1024;
+
+                string destinationFilePath = null;
+                try
+                {
+                    destinationFilePath = CreateFileWithRandomContent(FileSize);
+
+                    static string CreateFileWithRandomContent(int fileSize)
+                    {
+                        string filePath = Path.GetTempFileName();
```

Comment:
aside, better to use the TempFile class, which allows the using pattern instead of writing out try/finally. It also avoids the problematic Path.GetTempFileName() which fails if someone else has leaked temp files.
https://github.com/dotnet/runtime/blob/19eb98df125a6bc766acdb8aacfc36c7d2a75403/src/libraries/Common/tests/System/IO/TempFile.cs#L22

Reply:
Oh ok, I'll fix it up in a follow-up PR